perf_benchmark/benchmark_main.cc

```diff
@@ -410,7 +410,7 @@ void NestedPayloadFromGrpc(::benchmark::State& state, uint64_t layers,
                            bool streaming, uint64_t stream_size,
                            absl::string_view msg_type) {
   // GetNestedPayload transfers the pointer ownership to the caller.
-  // We wrap the pointer with unique_ptr to manage the pointer.
+  // We wrap the pointer with unique_ptr to manage the pointer lifecycle.
```

Comment:
// We wrap ..

This comment make sense only if `GetNestedPayload` return a raw point.  If so, maybe  change it to return unique_ptr.

Reply:
The reason why I can't return a unique_ptr is that `GetNestedPayload` struct the `NestedPayload` proto object recursively.  
The auto-generated protobuf object has a method named `set_allocated_nested()` which according to the documentation `sets the string object to the field and frees the previous field value if it exists`. This means it expects a raw pointer and will take on the pointer's ownership. I can only pass in raw pointer to it.

But I just realized I can wrap the recursive call with another function :)

Reply:
Updated again to just use unique_ptr. I just learned I can call unique_ptr#release() to explicitly transfer the pointer ownership.